core/vapp-python/.project-creation/run.py

```diff
@@ -32,9 +37,26 @@ def verbose_copy(src, dst) -> object:
 
 def clone_sdk(sdk_temp_dir: str) -> None:
     with open(f"{os.path.dirname(__file__)}/config.json") as f:
-        repo_url = json.load(f)["sdkUri"]
+        config = json.load(f)
+        repo_url = config["sdkUri"]
+        repo_version = config["sdkVersion"]
         try:
-            subprocess.check_call(["git", "clone", repo_url, sdk_temp_dir, "--quiet"])
+            clean_up_sdk_temp()
+            subprocess.check_call(
```

Comment:
For better testability (read: mockability) you should do the actual `subprocess.check_call` in a separate function. i.e. `_invoke_git_shallow_clone_branch(...)`